src/models/CourseHistory.js

```diff
@@ -0,0 +1,18 @@
+const mongoose = require('mongoose');
+const { SLOT_CREATION } = require('../helpers/constants');
+const addressSchemaDefinition = require('./schemaDefinitions/address');
+
+const ACTION_TYPES = [SLOT_CREATION];
+
+const CourseHistorySchema = mongoose.Schema({
+  createdBy: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true, immutable: true },
+  action: { type: String, required: true, enum: ACTION_TYPES, immutable: true },
+  course: { type: mongoose.Schema.Types.ObjectId, ref: 'Course', required: true, immutable: true },
+  slot: {
+    startDate: { type: Date, required: () => this.action === SLOT_CREATION },
```

Comment:
je mettrai plutot `[SLOT_CREATION].includes(this.action)` car ce sera aussi obligatoire pour `SSLOT_DELETION` par exemple donc on peut anticiper ;)

Reply:
bien vu ! done